stdlib/public/core/StringLegacy.swift

```diff
@@ -34,13 +34,19 @@ extension String {
       return
     }
 
-    // TODO(String performance): We can directly call appendInPlace
-    var result = String()
-    result.reserveCapacity(repeatedValue._guts.count &* count)
-    for _ in 0..<count {
-      result += repeatedValue
+	var repeatedValue = repeatedValue
+	self.init(_uninitializedCapacity: repeatedValue._guts.count &* count) {
+	  buffer in
+	  repeatedValue.withUTF8 { repeatedValueUTF8 in
+	    let repeatedCount = repeatedValueUTF8.count
+	    for i in 0..<count {
+          let range = i * repeatedCount ..< (i + 1) * repeatedCount
+            _ = UnsafeMutableBufferPointer(rebasing: buffer[range])
+              .initialize(from: repeatedValueUTF8)
+        }
```

Comment:
Would this approach cause a different result than does the original implementation when we have Unicode sequences that combine on concatenation?

Reply:
Unfortunately, yes — while you can use the length-count product as a capacity for the resulting string, the actual new count (returned below) needs to take into account combinations. A simple test for this is the string `"\u{301}cafe"`. That's 5 characters, but the result of `String(repeating: "\u{301}cafe", count: 5)` has a count of only 21.